src/components/Nav/Nav.js

```diff
@@ -1,12 +1,22 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Link, useLocation } from 'react-router-dom';
 import styled from 'styled-components';
 import TopBanner from './TopBanner';
 import { flexSet } from '../../styles/Variable';
 
-function Nav(props) {
+function Nav() {
   const location = useLocation();
 
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
+  useEffect(() => {
+    if (localStorage.getItem('access_token')) {
+      setIsLoggedIn(true);
+    } else {
+      setIsLoggedIn(false);
+    }
+  }, []);
```

Comment:
useEffect로 컴포넌트가 마운트 됬을때 판단하는 것 보다 useState의 초기값자리에 바로 넣어주시는 게 좋아보입니다!
그리고 state를 주기적으로 바꾸는 게 아니라 단순히 매 렌더링때 마다 로그인 되어있는지 여부 판단하려면은 굳이 state로 관리하지 않아도 되어 보이고 일반 변수로 선언해도 되어 보입니다!
```suggestion
const isLogined = localStorage.getItem("access_token");
```

Reply:
함수형 컴포넌트안에서 작성한 로직은 매 렌더링때마다 수행되니까

매 렌더링때마다 로컬스토리지에 있는 토큰값이 있는지 확인하고 그걸 isLogined 변수에 할당 -> 그 값을 이용해서 로그인 or 로그아웃 텍스트 표시로 동작합니다!

좀 더 연산을 줄이려면 기존에 하신 것처럼 state로 로그인 여부를 판단할 수도 있는데
이경우에는 전역상태관리를 통해서 로그인 됬을때 해당 state를 true로 바꾸고, 로그아웃했을때 해당 state를 false로 바꾸는 등이 동작이 추가될 것 같아서 전역상태관리를 지금 도입하기에는 무리가 좀 있어보여서 첫번째 방법대로 하시는게 좋을 것 같아보입니다!

조금이라도 이해 안되시는 부분 있으면 와서 질문주세요!